src/components/ModalConfirmCode.tsx

```diff
@@ -0,0 +1,108 @@
+import { connected } from 'process';
+import React,  {FC, useEffect, useState} from 'react';
+import './ModalConfirmCode.scss';
+
+interface CardProps {
+  active: boolean;
+  setActive: (bool: boolean) => void ;
+}
+
+const ModalConfirmCode: FC<CardProps> = 
+  ({
+    active, 
+    setActive,
+  }) => {
+  
+  const [code, setCode] = useState<string>('');
```

Comment:
const [code, setCode] = useState<string>('');

No need to write here type _string_ as you set default value '' (empty string). Typescript can infer the type

https://www.typescriptlang.org/docs/handbook/type-inference.html

THe same remark for 
  const [codeDirty, setCodeDirty] = useState<boolean>(false);
  const [codeError, setCodeError] = useState<string>('The input field must be filled');
  const [formValid, setFormValid] = useState<boolean>(false);